ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -745,17 +742,21 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = do
               -- Clear the tentative header
               prevTentativeHeader <- swapTVar varTentativeHeader SNothing
 
-              -- Update the followers
-              --
-              -- 'Follower.switchFork' needs to know the intersection point
-              -- (@ipoint@) between the old and the current chain.
-              let ipoint = castPoint $ Diff.getAnchorPoint chainDiff
-              followerHandles <- Map.elems <$> readTVar cdbFollowers
-              forM_ followerHandles $ \followerHandle ->
-                fhSwitchFork followerHandle ipoint newChain
+              when (chainSwitchType == SwitchingToAFork) $ do
```

Comment:
I would rather have a `case` with exhaustive explicit alternatives (mostly trivial), so that any changes to the data type will necessitate intentional (probably trivial) changes here.